src/main/java/com/google/devtools/build/lib/syntax/UnaryOperatorExpression.java

```diff
@@ -63,29 +63,31 @@ private static Object evaluate(TokenKind op, Object value, Location loc)
         return !EvalUtils.toBoolean(value);
 
       case MINUS:
-        if (!(value instanceof Integer)) {
-          throw new EvalException(
-              loc,
-              String.format(
-                  "unsupported operand type for -: '%s'", EvalUtils.getDataTypeName(value)));
+        if (value instanceof Integer) {
+          try {
+            return Math.negateExact((Integer) value);
+          } catch (ArithmeticException e) {
+            // Fails for -MIN_INT.
+            throw new EvalException(loc, e.getMessage());
+          }
         }
-        try {
-          return Math.negateExact((Integer) value);
-        } catch (ArithmeticException e) {
-          // Fails for -MIN_INT.
-          throw new EvalException(loc, e.getMessage());
+
```

Comment:
You need a 'break' here. (I'm surprised you don't get an "unintended fall-through" warning or error at compile time.)